src/Common/ProcessExecutor.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Robo\Common;
+
+use Robo\Contract\OutputAdapterInterface;
+use Robo\Contract\OutputAwareInterface;
+use Robo\Contract\TaskInterface;
+use Robo\Contract\VerbosityThresholdInterface;
+use Robo\Robo;
+use Robo\Task\BaseTask;
+
+class ProcessExecutor implements OutputAwareInterface
+{
+    use ExecTrait;
+    use TaskIO; // uses LoggerAwareTrait and ConfigAwareTrait
+    use ProgressIndicatorAwareTrait;
+    use OutputAwareTrait;
+
+    public function __construct($process)
+    {
+        $this->process = $process;
+        $this->logger = Robo::logger();
+        $this->setOutputAdapter(new OutputAdapter());
+        $this->outputAdapter()->setOutput(Robo::output());
```

Comment:
Regarding the static methods used here, things like `Robo::log()` and `Robo::output()` are provided for the convenience of Robo users who just need access to these objects without doing a lot of work with dependency injection. I prefer they not be used within any of the central parts of the Robo framework. There may be some that still remain, but these are deprecated and to be removed (e.g. the reference to Robo::logger() in the TaskIO trait).

I'd be satisfied to consider `ProcessExecutor` to be similar to these static constructors -- that is, something provided primarily for the convenience of Robo clients, but not itself used by Robo. In this context, the use of the static service provider methods is fine with me, as long as ProcessExecutor itself is not used by Robo internal functions (which should always be tasks that use ExecTrait with proper DI). However, you should definitely allow the logger and the output interface to be passed in as optional parameters to the constructor, and provide `Robo::logger()` and `Robo::output()` if these are omitted.